### PR TITLE
bugfix:  fix the NPE and reduce the request when lockkey is null

### PR DESCRIPTION
**Pull request by lightClouds917**
### Ⅰ. Describe what this PR did
1.fix: Fix the NPE when lockkey is null;
2.optimize:  Reduce once db request when lockkey is null;

### Ⅱ. Does this pull request fix one issue?

yes.
[issue](https://github.com/seata/seata/issues/2496)

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java [hunk 1]
```diff
@@ -198,8 +200,10 @@ private void doCommit() throws SQLException {
     }
 
     private void processLocalCommitWithGlobalLocks() throws SQLException {
-
-        checkLock(context.buildLockKeys());
+        Set<String> lockKeys = context.getLockKeysBuffer();
+        if (CollectionUtils.isNotEmpty(lockKeys)) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java:204]: Is it more appropriate to put logic with lockkey as empty in the checkLock method?

Comment (lightClouds917) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java:204]: ok,i will change

File: server/src/main/java/io/seata/server/lock/AbstractLockManager.java [hunk 2]
```diff
@@ -153,6 +153,9 @@ protected Locker getLocker() {
                                             Long branchID) {
         List<RowLock> locks = new ArrayList<RowLock>();
 
+        if (StringUtils.isEmpty(lockKey)) {
+            return null;
+        }
```
Comment (zjinlei) [inline server/src/main/java/io/seata/server/lock/AbstractLockManager.java:158]: Incorrect

Comment (slievrly) [reply server/src/main/java/io/seata/server/lock/AbstractLockManager.java:158]: check  `isLockable` method.

File: server/src/main/java/io/seata/server/lock/AbstractLockManager.java [hunk 3]
```diff
@@ -153,6 +153,9 @@ protected Locker getLocker() {
                                             Long branchID) {
         List<RowLock> locks = new ArrayList<RowLock>();
 
+        if (StringUtils.isEmpty(lockKey)) {
```
Comment (l81893521) [inline server/src/main/java/io/seata/server/lock/AbstractLockManager.java:156]: That s ok. And I think can remove the code
```
if (StringUtils.isNullOrEmpty(lockKey)) {
            // no lock
            return true;
}
```
at 
```
acquireLock(BranchSession branchSession)
```

Comment (lightClouds917) [reply server/src/main/java/io/seata/server/lock/AbstractLockManager.java]: `isLockable `will call this also，and it has no NEP judgement;

File: rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java (hunk 1 above)
Comment (Garlichead) [reply rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java]: image
the master code

**Pull request merged**